apps/events/views.py

```diff
@@ -0,0 +1,65 @@
+from django.contrib.auth.mixins import LoginRequiredMixin, PermissionRequiredMixin
+from django.shortcuts import get_object_or_404, redirect, render
+from django.urls import reverse_lazy
+from django.views.generic import CreateView, DeleteView, UpdateView
+
+from apps.events.forms import EventForm
+from apps.events.models import Event
+from cosmos.settings import LOGIN_URL
+
+
+def events_list(request):
+    if not request.user.is_authenticated:
+        events_list = Event.objects.filter(member_only=False).order_by("-start_time")
+    else:
+        events_list = Event.objects.order_by("start_time").all()
+    context = {
+        "events_list": events_list,
+    }
+    return render(request, "events/events_list.html", context)
+
+
+def event_view(request, pk):
+    event = get_object_or_404(Event, pk=pk)
+    context = {"event": event}
+    if event.member_only and not request.user.is_authenticated:
+        return redirect("%s?next=%s" % (LOGIN_URL, request.path))
```

Comment:
Why redirect? Should it not be permission denied? Maybe we want the permission denied screen to redirect to login screen with a banner on top that says you dont have permission to view that object or sth? Seems like the cleaner solution.

Reply:
Sounds like a good idea, I'm not quite sure how to do that though, for now I'm just using PermissionDenied.
Just for reference, the quoted code is also used for the news view